src/controller_examples/rabbitmq_controller/proof/safety.rs

```diff
@@ -19,4 +19,25 @@ use crate::rabbitmq_controller::{
 use crate::temporal_logic::{defs::*, rules::*};
 use vstd::prelude::*;
 
-verus! {}
+verus! {
+
+spec fn stateful_set_not_scaled_down(rabbitmq: RabbitmqClusterView, msg: Message) -> StatePred<RMQCluster> {
+    |s: RMQCluster| {
+        let sts_key = make_stateful_set_key(rabbitmq.object_ref());
+        &&& s.message_in_flight(msg)
+        &&& msg.dst.is_KubernetesAPI()
+        &&& msg.content.is_update_request()
+        &&& msg.content.get_update_request().key == sts_key
```

Comment:
We should also say the msg won't be rejected by the k8s api server (validate_update_request returns true). We don't need to worry about the invalid request, which won't affect the cluster state anyway